src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PackageCodeAggregator.cs

```diff
@@ -144,17 +149,29 @@ internal static class PackageCodeAggregator
 
                 mostRecentPackageCodeHistory[change.Key] = packageCodeHistory;
 
-                packageCodeAggregate.PackageCodeIds.Add(packageCodeHistory.Id);
-            }
-
-            if (change.Value.PackageCodeVmType != null)
-            {
-                packageCodeHistory.VmType = change.Value.PackageCodeVmType.Value.VmType.ToModel();
-            }
-
-            if (change.Value.PackageCodeOriginalCode != null)
-            {
-                packageCodeHistory.Code = change.Value.PackageCodeOriginalCode.Value.CodeHex.ConvertFromHex();
+                if (change.Value is { PackageCodeIsDeleted: true, CodeVmTypeIsDeleted: true })
```

Comment:
Could we please create regular method or extension method with a simple name "should be deleted"? It could throw on ambiguous data. Pattern matching can be abused so easily ;-)